src/indexes/text/fuzzy.h

```diff
@@ -2,42 +2,168 @@
  * Copyright (c) 2025, valkey-search contributors
  * All rights reserved.
  * SPDX-License-Identifier: BSD 3-Clause
- *
  */
 
 #ifndef _VALKEY_SEARCH_INDEXES_TEXT_FUZZY_H_
 #define _VALKEY_SEARCH_INDEXES_TEXT_FUZZY_H_
 
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include "absl/container/inlined_vector.h"
+#include "absl/strings/string_view.h"
+#include "posting.h"
+#include "radix_tree.h"
+
 namespace valkey_search::indexes::text {
 
-//
-// Fuzzy iterator that returns keys
-//
-// two stage iterator. Does FuzzyWordIterator then maps that into keys
-//
-struct FuzzyKeyIterator : public public indexes::EntriesFetcherIteratorBase {
-  FuzzyIterator(indexes::Text& index, absl::string_view word, size_t distance);
-  virtual bool Done() const override;
-  virtual void Next() = override;
-  virtual const Key& operator*() const override;
-  virtual std::unique_ptr<WordIterator> clone() const override;
-};
+// Fuzzy search using Damerau-Levenshtein distance on RadixTree
+template <typename Target>
```

Comment:
I'm not sure this needs to be templated. The return type already exposes that a postings object is in the tree `std::vector<Postings::KeyIterator>`. I guess if we have different pointer types in trees pointing to postings objects at some point it could theoretically be useful, but I'm not sure why fuzzy search would operate on multiple different types of trees.